### PR TITLE
Auto Query Fixer: Implement TableNotFoundFixer

### PR DESCRIPTION
**Pull request by mingen-pan**
Implement `TableNotFoundFixer` and `FixerFactory`.

Test is included.

### CODE REVIEW

File: tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/entity/FixOption.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.google.cloud.bigquery.utils.queryfixer.entity;
+
+import lombok.Value;
+
+@Value(staticConstructor = "of")
+public class FixOption {
+  /**
+   * A short description about this fixing option. Usually, if an identifier is to be modified, this
+   * field will be the new identifier to replace. For example, if a table is to be replaced, this
```
Comment (kikkyo) [inline tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/entity/FixOption.java:9]: nit: stylewise we usually don't use paragraph for field comment.
Just use "//..." for short comment.

File: tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/fixer/TableNotFoundFixer.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    List<String> tableNames =
+        bigQueryService.listTableNames(fullTableId.getProject(), fullTableId.getDataset());
+
+    StringUtil.SimilarStrings similarTables =
+        StringUtil.findSimilarWords(tableNames, fullTableId.getTable());
+
+    // this is an arbitrary standard. It requires the candidate table should share at least 50%
+    // similarity as
+    // the incorrect table typo.
+    int editDistanceThreshold = (fullTableId.getTable().length() + 1) / 2;
```
Comment (kikkyo) [inline tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/fixer/TableNotFoundFixer.java:53]: maybe add a TODO to make this user configurable.

File: tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/fixer/TableNotFoundFixer.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+    if (similarTables.getStrings().isEmpty()
+        || similarTables.getDistance() > editDistanceThreshold) {
+      return FixResult.failure(err);
+    }
+
+    // this method only finds the first occurrence of the incorrect table. It is possible that this
```
Comment (kikkyo) [inline tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/fixer/TableNotFoundFixer.java:60]: nit: This

File: tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/fixer/TableNotFoundFixer.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    String tableName = contents.get(3);
+    return TableId.of(projectId, datasetId, tableName);
+  }
+
+  private String constructFullTableName(String projectId, String datasetId, String tableName) {
+    return String.format("%s.%s.%s", projectId, datasetId, tableName);
+  }
+
+  private int findTheIndexOfIncorrectTable() {
+    // the table in the error message is presented in the legacySQL mode, but this fixer is used to
```
Comment (kikkyo) [inline tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/fixer/TableNotFoundFixer.java:103]: So we rely on the error message to identify if they used legacy sql?

Comment (mingen-pan) [reply tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/fixer/TableNotFoundFixer.java:103]: Here I mean the error message represents the table in legacySQL format, which is `project:dataset.table`. I guess we could be an issue for BigQuery to update this error message.

File: tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/fixer/TableNotFoundFixer.java (hunk 3 above)
Comment (mingen-pan) [reply tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/fixer/TableNotFoundFixer.java:60]: Done

File: tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/fixer/TableNotFoundFixer.java (hunk 2 above)
Comment (mingen-pan) [reply tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/fixer/TableNotFoundFixer.java:53]: Done

File: tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/entity/FixOption.java (hunk 1 above)
Comment (mingen-pan) [reply tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/entity/FixOption.java:9]: Done

**Pull request merged**